### ISSUE TITLE
IPv6 Local-Link oddities under LX

### ISSUE THREAD

**Issue by sjorge** [joyent/illumos-joyent#78]
```
root@artemis:/run/plex/logs# ip a
1: lo: <LOOPBACK,MULTICAST,UP> mtu 8232
    link/loopback 00:00:00:00:00:00
    inet 127.0.0.1/8 scope global dynamic
    inet6 ::1/128 scope global dynamic
3: eth0: <BROADCAST,MULTICAST,UP> mtu 1500 qlen 1
    link/ether 00:15:00:05:d2:a4
    inet 172.16.10.150/24 scope global dynamic
    inet6 fe80::215:ff:fe05:d2a4/10 scope global dynamic
    inet6 2001:6f8:1480:10::150/64 scope global dynamic
3: eth0:1: <MULTICAST,UP> mtu 1500 qlen 1
    link/ether 00:15:00:05:d2:a4
    inet 172.16.10.150/24 scope global dynamic
    inet6 fe80::215:ff:fe05:d2a4/10 scope global dynamic
    inet6 2001:6f8:1480:10::150/64 scope global dynamic
root@artemis:/run/plex/logs# ping fe80::215:ff:fe05:d2a4
ping: unknown host fe80::215:ff:fe05:d2a4
root@artemis:/run/plex/logs# ping6 fe80::215:ff:fe05:d2a4
WARNING: your kernel is veeery old. No problems.
PING fe80::215:ff:fe05:d2a4(fe80::215:ff:fe05:d2a4) 56 data bytes
ping: recvmsg: Operation not supported
```

```SmartOS Live Image v0.147+ build: 20151001T070028Z``` (I reverted to stock PI to make sure it was not my global zone ipv6 changes causing this)

**Comment by sjorge** [comment]
@pfmooney here is the issue you asked for on IRC

I think Plex's last error about not being able to reach all discovered networks is related to this.
Disabling IPv6 support in plex makes the error go away, with it enabled I the following error a few times per minute:

```Oct 10, 2015 17:27:59 [0x7fffe5e00700] WARN - NetworkServiceBrowser: Error sending out discover packet: send_to: Network is unreachable```

So I started checking my IPv6 networks, global one 2001:6f8:1480:10::/64 works fine and the box is reachable on it, not so on fe80::/10

Even pinging it's own address fails? :question:

**Comment by wiedi** [comment]
does ping work if you specify the interface?

**Comment by sjorge** [comment]
That didn't make a difference:
```
root@artemis:~# ping6  -I eth0 fe80::215:ff:fe05:d2a4
ping6: Warning: source address might be selected on device other than eth0.
WARNING: your kernel is veeery old. No problems.
PING fe80::215:ff:fe05:d2a4(fe80::215:ff:fe05:d2a4) from fe80::215:ff:fe05:d2a4 eth0: 56 data bytes
ping: recvmsg: Operation not supported
ping: recvmsg: Operation not supported
```

```
root@artemis:~# ip a
1: lo: <LOOPBACK,MULTICAST,UP> mtu 8232
    link/loopback 00:00:00:00:00:00
    inet 127.0.0.1/8 scope global dynamic
    inet6 ::1/128 scope global dynamic
3: eth0: <BROADCAST,MULTICAST,UP> mtu 1500 qlen 1
    link/ether 00:15:00:05:d2:a4
    inet 172.16.10.150/24 scope global dynamic
    inet6 fe80::215:ff:fe05:d2a4/10 scope global dynamic
    inet6 2001:6f8:1480:10::150/64 scope global dynamic
3: eth0:1: <MULTICAST,UP> mtu 1500 qlen 1
    link/ether 00:15:00:05:d2:a4
    inet 172.16.10.150/24 scope global dynamic
    inet6 fe80::215:ff:fe05:d2a4/10 scope global dynamic
    inet6 2001:6f8:1480:10::150/64 scope global dynamic
root@artemis:~# ifconfig
eth0      Link encap:Ethernet  HWaddr 00:15:00:05:d2:a4
          inet addr:172.16.10.150  Bcast:172.16.10.255  Mask:255.255.255.0
          inet6 addr: fe80::215:ff:fe05:d2a4/10 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:2105167 errors:0 dropped:0 overruns:0 frame:0
          TX packets:21650732 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:231140872 (231.1 MB)  TX bytes:30624295308 (30.6 GB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Link
          UP LOOPBACK RUNNING MULTICAST  MTU:8232  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
root@artemis:~# /native/usr/sbin/ifconfig
lo0: flags=2001000849<UP,LOOPBACK,RUNNING,MULTICAST,IPv4,VIRTUAL> mtu 8232 index 1
        inet 127.0.0.1 netmask ff000000
eth0: flags=201000843<UP,BROADCAST,RUNNING,MULTICAST,IPv4,CoS> mtu 1500 index 3
        inet 172.16.10.150 netmask ffffff00 broadcast 172.16.10.255
        ether 0:15:0:5:d2:a4
lo0: flags=2002000849<UP,LOOPBACK,RUNNING,MULTICAST,IPv6,VIRTUAL> mtu 8252 index 1
        inet6 ::1/128
eth0: flags=20202000841<UP,RUNNING,MULTICAST,IPv6,CoS> mtu 1500 index 3
        inet6 fe80::215:ff:fe05:d2a4/10
        ether 0:15:0:5:d2:a4
eth0:1: flags=20202000841<UP,RUNNING,MULTICAST,IPv6,CoS> mtu 1500 index 3
        inet6 2001:6f8:1480:10::150/64
```

**Comment by sjorge** [comment]
Also opening up the network by allowing all traffic on the vnic results in the same.

**Comment by melloc** [comment]
This should be fixed in the next release by 4e92ac8.

**Comment by sjorge** [comment]
Cool that ping will work now! A while ago pmooney helped me track down the plex problem to a difference in loopback implementation between illumos and linux though.
